src/appleseed/foundation/utility/xmlelement.h

```diff
@@ -156,13 +156,14 @@ inline void XMLElement::write(const ContentType content_type)
     assert(!m_is_open);
 
     // Open the tag.
-    std::fprintf(m_file, "%s<%s", m_indenter.c_str(), m_name.c_str());
+    std::fprintf(m_file, "%s<%s", m_indenter.c_str(), replace_special_xml_characters(m_name).c_str());
```

Comment:
The name of the tag certainly shouldn't contain any special characters. I would suggest to revert this.

Reply:
I thought about it too. In the end of the day, if there are any special characters in the tag, we got problems anyways, but I thought it would be more thorough to do it that way.

Reply:
Less code is always better (code is a liability, not an asset) in every dimension: speed, compactness, readability, etc. Since we know per the XML spec that tags cannot contain special characters, we definitely don't want to do any escaping which involves memory allocations, memory fragmentation, string copies, etc.